File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第7章：形态学处理/7.1/erode.py
# -*- coding: utf-8 -*-
import sys
import cv2
#主函数
if __name__ =="__main__":
    if len(sys.argv) > 1:
    #第一步：读入图像
        I = cv2.imread(sys.argv[1],cv2.IMREAD_GRAYSCALE)
    else:
        print "Usge:python erode.py imageFile"
    #创建结构元
    s = cv2.getStructuringElement(cv2.MORPH_RECT,(5,5))
    #腐蚀图像
    r = cv2.erode(I,s)
    #显示原图和腐蚀后的结果
    cv2.imshow("I",I)
    cv2.imshow("erode",r)
    cv2.imwrite("erode.jpg",r)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第7章：形态学处理/7.1/erode.cpp
#include<opencv2/core/core.hpp>
#include<opencv2/highgui/highgui.hpp>
#include<opencv2/imgproc/imgproc.hpp>
using namespace cv;
int mian(int argc, char*argv[])
{
	//输入图像
	Mat I = imread(argv[1], CV_LOAD_IMAGE_GRAYSCALE);
	if (!I.data)
		return -1;
	//创建矩形结构元
	Mat s = cv::getStructuringElement(cv::MORPH_RECT, Size(3, 3));
	//腐蚀图像
	Mat E;
	cv::erode(I, E, s,Point(-1,-1),2);
	//显示图像
	imshow("I", I);
	imshow("erode", E);
	waitKey(0);
	return 0;
}